truffleruby/src/main/java/org/truffleruby/core/CoreLibrary.java

```diff
@@ -1099,6 +1099,10 @@ public static boolean fitsIntoInteger(long value) {
         return ((int) value) == value;
     }
 
+    public static boolean fitsIntoUnsignedInteger(long value) {
+        return value == (value & 0xffffffffl) || value < 0 && value >= Integer.MIN_VALUE;
```

Comment:
This seems a bit strange of a condition. Should a small negative long "fit" in a unsigned integer?
Could the condition be reduced to just 2 comparisons?